### PR TITLE
Enable FTD for OpenThread connection

### LINKED ISSUE #49: Investigate using Mesh OT
Investigate if it's possible to use Mesh Network for OT

### PR DESCRIPTION
**Pull request by jcorde**
- Move OT-Storage sub-partition to Scratch area.
 - Allow UART usage for Dongle
 - Set OpenThread credentials on Main app initialization.
 - Enable router and leader roles.

Closes #49

### CODE REVIEW

File: core/src/ot_config.c [hunk 1]
```diff
@@ -233,7 +234,8 @@ int ot_config_stop(void)
 bool ot_config_is_ready(void)
 {
 	/* Device is ready when its role is set to Child */
```
Comment (ramonhpr) [inline core/src/ot_config.c]: Change the comment.

Comment (ramonhpr) [reply core/src/ot_config.c]: Now it can be ready as router and leader too

**Review by ramonhpr — CHANGES_REQUESTED** [review]

File: core/src/ot_config.c (hunk 1 above)
Comment (jcorde) [reply core/src/ot_config.c]: Done

**Review by ramonhpr — APPROVED** [review]

File: setup/boards/nrf52840_pca10059.conf [hunk 2]
```diff
@@ -1,2 +1,2 @@
-# Using first 2 flash areas from storage slot for Settings subsystem
-CONFIG_SETTINGS_FCB_NUM_AREAS=2
+# Using all 4 flash areas from storage slot for Settings subsystem
```
Comment (alla3) [inline setup/boards/nrf52840_pca10059.conf]: I suggest you leave **pages** or **areas** in both DTS files.

File: core/boards/nrf52840_pca10056-sub-slots-debug.dts [hunk 3]
```diff
@@ -29,12 +29,13 @@
 		};
 
 		/*
-		 * Using last 4 pages of storage partition
+		 * Using 4 pages of scratch partition for OpenThread storage.
```
Comment (alla3) [inline core/boards/nrf52840_pca10056-sub-slots-debug.dts]: I suggest you specify which pages the scratch is using.

File: core/boards/nrf52840_pca10056-sub-slots-stock.dts [hunk 4]
```diff
@@ -28,12 +28,13 @@
 		};
 
 		/*
-		 * Using last 4 pages of storage partition
+		 * Using 4 pages of scratch partition for OpenThread storage.
```
Comment (alla3) [inline core/boards/nrf52840_pca10056-sub-slots-stock.dts]: I suggest you specify which pages the scratch is using.

File: core/boards/nrf52840_pca10059-sub-slots-debug.dts [hunk 5]
```diff
@@ -29,9 +29,11 @@
 		};
 
 		/*
-		 * Using last 2 pages of storage partition
+		 * Using 4 pages of scratch partition for OpenThread storage.
```
Comment (alla3) [inline core/boards/nrf52840_pca10059-sub-slots-debug.dts]: I suggest you specify which pages the scratch is using.

File: core/boards/nrf52840_pca10059-sub-slots-stock.dts [hunk 6]
```diff
@@ -27,9 +27,14 @@
 			reg = <0x00030000 0x00003E000>;
 		};
 
-		ot_partition: partition@fe000 {
+		/*
+		 * Using 4 pages of scratch partition for OpenThread storage.
```
Comment (alla3) [inline core/boards/nrf52840_pca10059-sub-slots-stock.dts]: I suggest you specify which pages the scratch is using.

File: core/boards/nrf52840_pca10056-sub-slots-debug.dts (hunk 3 above)
Comment (jcorde) [reply core/boards/nrf52840_pca10056-sub-slots-debug.dts]: Done

File: core/boards/nrf52840_pca10056-sub-slots-stock.dts (hunk 4 above)
Comment (jcorde) [reply core/boards/nrf52840_pca10056-sub-slots-stock.dts]: Done

File: core/boards/nrf52840_pca10059-sub-slots-debug.dts (hunk 5 above)
Comment (jcorde) [reply core/boards/nrf52840_pca10059-sub-slots-debug.dts]: Done

File: core/boards/nrf52840_pca10059-sub-slots-stock.dts (hunk 6 above)
Comment (jcorde) [reply core/boards/nrf52840_pca10059-sub-slots-stock.dts]: Done

File: setup/boards/nrf52840_pca10059.conf (hunk 2 above)
Comment (jcorde) [reply setup/boards/nrf52840_pca10059.conf]: Done

Comment (jcorde) [reply setup/boards/nrf52840_pca10059.conf]: Done

**Review by alla3 — APPROVED** [review]

**Pull request merged**